storage/google/cloud/storage/client.py

```diff
@@ -202,28 +202,52 @@ def batch(self):
         """
         return Batch(client=self)
 
-    def get_bucket(self, bucket_name):
-        """Get a bucket by name.
+    def get_bucket(self, bucket_or_name):
+        """API call: retrieve a bucket via a GET request.
 
-        If the bucket isn't found, this will raise a
-        :class:`google.cloud.exceptions.NotFound`.
+        See
+        https://cloud.google.com/storage/docs/json_api/v1/buckets/get
 
-        For example:
+        Args:
+            bucket_or_name (Union[ \
+                :class:`~google.cloud.storage.bucket.Bucket`, \
+                 str, \
+            ]):
+                The bucket resource to pass or name to create.
 
-        .. literalinclude:: snippets.py
-            :start-after: [START get_bucket]
-            :end-before: [END get_bucket]
+        Returns:
+            google.cloud.storage.bucket.Bucket
+                The bucket matching the name provided.
 
-        This implements "storage.buckets.get".
+        Raises:
+            google.cloud.exceptions.NotFound
+                If the bucket is not found.
 
-        :type bucket_name: str
-        :param bucket_name: The name of the bucket to get.
+        Examples:
+            Retrieve a bucket using a string.
+
+            .. literalinclude:: snippets.py
+                :start-after: [START get_bucket]
+                :end-before: [END get_bucket]
+
+            Get a bucket using a resource.
+
+            >>> from google.cloud import storage
+            >>> client = storage.Client()
+
+            >>> # Set properties on a plain resource object.
+            >>> bucket = storage.Bucket("my-bucket-name")
```

Comment:
I fear that this example could lead to people thinking they have to create a Bucket object when they can just pass in a string.

This implementation is more to provide the same mechanism as `reload()` is currently used, so we should indicate that somehow you got a bucket but think it might be out of date and want to reload it.

Maybe change this line to `get_bucket("my-bucket-name")`, and then between this and the next call to `get_bucket`, add a comment like `# Time passes. Another program may have modified the bucket in the meantime, so you want to get the latest state.`

Reply:
Modified the example as directed!